code/game/magic/Uristrunes.dm

```diff
@@ -147,36 +161,104 @@ var/list/uristrune_cache = list()
 					var/sw = I.GetPixel(x - 1, y - 1)
 
 					if(ne == "#000000" || se == "#000000" || nw == "#000000" || sw == "#000000")
-						I.DrawBox(rgb(200, 0, 0, 100), x, y)
+						I.DrawBox(bc2, x, y)
 
 	var/icon/result = icon(I, "")
 
-	result.Insert(I,  "", frame = 1, delay = 10)
+	I.MapColors(0.5,0,0,0,0.5,0,0,0,0.5)//we'll darken that color a bit
+
+	var/icon/I1 = icon(I, "")
+	I1.MapColors(1,0.05,0,0,1,0.05,0.05,0,1)
+	result.Insert(I1,  "", frame = 1, delay = 10)
+
+/*
+	I.MapColors(rgb(0x80,0,0,0), rgb(0,0x80,0,0), rgb(0,0,0x80,0), rgb(0,0,0,0xff))//we'll darken that color a bit
+
+	var/icon/I1 = icon(I, "")
+	I1.MapColors(rgb(0xff,0x08,0,0), rgb(0,0xff,0x08,0), rgb(0x08,0,0xff,0), rgb(0,0,0,0xff))
+	result.Insert(I1,  "", frame = 1, delay = 10)
+*/
+
+	if(animated == 1)
+		var/icon/I2 = icon(I, "")
+		I2.MapColors(1.125,0.06,0,0,1.125,0.06,0.06,0,1.125)
```

Comment:
The real cost of animate is almost always the generation of appearances, which is an internal byond construct to describe how an icon looks when it renders on the map (you can see the same cost of appearance generation when adding tons of overlays to something). On top of that for animate what occurs is the interpolation of appearances, when you tell it to modify one value all the way to the other and it decides how its going to splice it up.

But using a pregenerated matrix of values with a repeating animation is actually very efficient and should have a negligible cost.